### DISCUSSION TITLE
Custom dynamic utility with nested values

### DISCUSSION THREAD

**Discussion by a-hirsch** [tailwindlabs/tailwindcss#10335] [Help] [answered]
Hey guys,
I have this config:
```js
module.exports = {
  content: ['./pages/**/*.{ts,tsx}', './components/**/*.{ts,tsx}'],
  theme: {
    gradientBorderPositions: {
      't': '100% 0 0 0',
      'r': '0 100% 0 0',
      'b': '0 0 100% 0',
      'l': '0 0 0 100%',
    },
  },
  plugins: [
    plugin(function ({ matchUtilities, theme }) {
      matchUtilities({
        'gradient-border': (position ) => ({
          borderImage: 'linear-gradient(to right, var(--tw-gradient-stops))',
          borderImageSlice: position,
          // borderWidth: position.replace('100%', width ?? '1px'),
          borderStyle: 'solid',
        }),
      }, {
        values: theme('gradientBorderPositions'),
      });
    }),
  ],
};
```
I wanted to create a utility for border image and make it similar to border classes, e.g.
```tsx
<div className="border-b-2" />
```
I can't get the `matchUtilities()` function accept more than 1 value, so I can do only
```tsx
<div className="gradient-border-b border-b-2" />
```
which is annoying, because I still have to add `border-b-2` class to show the border.
Is it possible to make it like `gradient-border-b-2`?

Thanks!

**Comment by wongjn** [comment, marked as answer]
There are only four edges, but an infinite amount of border widths possible, so consider having the edges be hardcoded in plugin and let the values be what is in configuration. This is akin to how core plugins work, for example, `borderWidth`: https://github.com/tailwindlabs/tailwindcss/blob/91b4b938665fa60b9a28543fda3b909d30a44857/src/corePlugins.js#L1487-L1505

Here is an example: https://play.tailwindcss.com/3BRxJGyL11?file=config

**Reply by a-hirsch** [reply]
Oh the example is perfect, tho I changed the borderWidth to be 
```
borderWidth: `${'0 '.repeat(i)}${borderWidth}${' 0'.repeat(3 - i)}`
```
So the border is only on one side
Thank you so much!
